### DISCUSSION TITLE
Get rate limit status for the Github Apps installed

### DISCUSSION THREAD

**Discussion by duckaraptor** [community/community#43215] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I want to know a way ( API Endpoint ) to get the rate limit status of the Github Apps with-in our organizations.
PS: I have tried looking it up in the docs, but there is a possibility I might have missed it.

**Comment by airtower-luna** [comment]
There is an endpoint for authenticated users: https://docs.github.com/en/rest/rate-limit?apiVersion=2022-11-28

In many cases it'll be simpler and more efficient to just look at the `x-ratelimit-*` headers in the response to *any* API request though, see: https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#checking-your-rate-limit-status

**Reply by duckaraptor** [reply]
Thanks for sharing the endpoint for authenticated users.
I also know from the GitHub docs that an GH app has a rate-limit of 15000 request per hour for GH Enterprise cloud.

I want to find the rate limit status exclusively for one GH app. If I use the above endpoint, It requires a Bearer token.
If I use a PAT for my account, then I get the rate-limits of my user.
If I use a client-secret or private key ( just in case) from the GH App, it does not work.

Any suggestions on what token to use for the rate-limit endpoint you mentioned about for a GitHub app.

**Reply by airtower-luna** [reply]
Then it's best to use the `x-ratelimit-*` headers, the documentation there explicitly mentions "number of API calls available to you or your app". :slightly_smiling_face:

**Comment by duckaraptor** [comment, marked as answer]
After juggling for two days, I was finally able to get the rate-limits for GH app.
It was my lack of understanding of GH Apps or Github Apps could have a better documentation, but here is how you  do it.
You need to authenticate as an installation for GH App ( https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-an-installation ) and then you can use 
```
curl -i -H "Authorization: Bearer <YOUR_INSTALLATION_ACCESS_TOKEN>" -H "Accept: application/vnd.github+json" https://api.github.com/rate_limit
```
